packages/web/src/pages/rewards-page/components/RewardPanel.tsx

```diff
@@ -0,0 +1,142 @@
+import { useFormattedProgressLabel } from '@audius/common/hooks'
+import {
+  ChallengeRewardID,
+  Name,
+  OptimisticUserChallenge
+} from '@audius/common/models'
+import {
+  ChallengeRewardsModalType,
+  challengesSelectors
+} from '@audius/common/store'
+import { isAudioMatchingChallenge } from '@audius/common/utils'
+import {
+  Box,
+  Flex,
+  IconCheck,
+  Paper,
+  ProgressBar,
+  Text,
+  useTheme
+} from '@audius/harmony'
+import { useSelector } from 'react-redux'
+
+import { make, track } from 'services/analytics'
+
+import { StatusPill } from './StatusPill'
+
+const { getOptimisticUserChallenges } = challengesSelectors
+
+type RewardPanelProps = {
+  title: string
+  description: (challenge?: OptimisticUserChallenge) => string
+  progressLabel?: string
+  remainingLabel?: string
+  openModal: (modalType: ChallengeRewardsModalType) => void
+  id: ChallengeRewardID
+}
+
+export const RewardPanel = ({
+  id,
+  title,
+  description,
+  openModal,
+  progressLabel,
+  remainingLabel
+}: RewardPanelProps) => {
+  const { color, spacing } = useTheme()
+  const userChallenges = useSelector(getOptimisticUserChallenges)
+
+  const openRewardModal = () => {
+    openModal(id)
+    track(
+      make({ eventName: Name.REWARDS_CLAIM_DETAILS_OPENED, challengeId: id })
+    )
+  }
+
+  const challenge = userChallenges[id]
+  const hasDisbursed = challenge?.state === 'disbursed'
+  const needsDisbursement = Boolean(challenge && challenge.claimableAmount > 0)
+  const shouldShowProgressBar =
+    challenge &&
+    challenge.max_steps &&
+    challenge.max_steps > 1 &&
+    challenge.challenge_type !== 'aggregate' &&
+    !hasDisbursed
+  const shouldShowNewChallengePill =
+    isAudioMatchingChallenge(id) && !needsDisbursement
+
+  const formattedProgressLabel: string = useFormattedProgressLabel({
+    challenge,
+    progressLabel,
+    remainingLabel
+  })
+
+  return (
+    <Paper
+      onClick={openRewardModal}
+      ph='unit2'
+      h={200}
+      flex={`0 0 calc(50% - ${spacing.unit4}px)`}
+      direction='column'
+      m='unit2'
+      shadow='flat'
+      border='strong'
+      css={{
+        minWidth: '336px',
+        backgroundColor: hasDisbursed ? color.neutral.n25 : undefined
+      }}
+    >
+      <Flex direction='column' h='100%'>
+        <Flex justifyContent='flex-end' mt='unit1' w='100%'>
```

Comment:
ditto here, feel like we should favor semantic spacing values (tho i always forget them too...)